File: 37- Longest consecutive sequence/Python/solution3.py
# Optimized brute force solution:

# Time complexity: O(n)
# Space complexity: O(n)

def longestConsecutiveSequence(arr):
  if len(arr) < 2:
    return len(arr)
  maxLength = 1
  values = set(arr)
  for element in values:
    if element-1 in values:
      continue
    else:
      right = element
      while right+1 in values:
        right += 1
      maxLength = max(maxLength, right-element+1)
  return maxLength

File: 37- Longest consecutive sequence/C++/solution3.cpp
// Optimized brute force solution:

// Time complexity: O(n)
// Space complexity: O(n)

int longestConsecutiveSequence(vector<int> arr){
  if(arr.size() < 2) return arr.size();
  int maxLength = 1;
  unordered_set<int> values(arr.begin(), arr.end());
  for(int element : values)
    if(values.find(element-1) != values.end()) continue;
    else{
      int right = element;
      while(values.find(right+1) != values.end()) right++;
      maxLength = max(maxLength, right-element+1);
    }
  return maxLength;
}


